phoenix-core/src/test/java/org/apache/phoenix/util/TestUtil.java

```diff
@@ -934,7 +936,35 @@ public IndexStateCheck(PIndexState indexState, Long indexDisableTimestamp, Boole
     		this.success = success;
     	}
     }
-    
+
+    public static void waitForIndexState(Connection conn, String fullIndexName, PIndexState expectedIndexState) throws InterruptedException, SQLException {
+        int maxTries = 60;
+        RetryCounter retrier = new RetryCounter(maxTries,
```

Comment:
@gokceni - Unfortunately, the RetryCounter is InterfaceAudience.Private in HBase, which means that outside projects such as Phoenix shouldn't depend on it.

Reply:
@swaroopak asked me to change the wait code to depend on it. Will role back.